### PR TITLE
MULE-16129: Cannot deactivate streaming in operations that return lists/maps of streams

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

File: core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java [hunk 1]
```diff
@@ -198,6 +204,32 @@ public void manage(InputStream stream, CoreEvent creatorEvent) {
...
+          && muleContext.getConfiguration().isInheritIterableRepeatability()) {
+        cursorStreamProviderFactory = forBytes().getNullCursorProviderFactory();
+      } else {
+        cursorStreamProviderFactory = forBytes().getDefaultCursorProviderFactory();
+      }
+    } else if (provider instanceof CursorStreamProviderFactory) {
+      cursorStreamProviderFactory = (CursorStreamProviderFactory) provider;
+      cursorIteratorProviderFactory = forObjects().getDefaultCursorProviderFactory();
+    } else {
+      cursorStreamProviderFactory = forBytes().getDefaultCursorProviderFactory();
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java]: IllegalArgumentException should be thrown here

File: core/src/main/java/org/mule/runtime/core/api/streaming/StreamingManager.java [hunk 2]
```diff
@@ -103,4 +106,13 @@
    */
   @Deprecated
   void manage(InputStream inputStream, CoreEvent creatorEvent);
+
+  /**
+   * @param provider the provider for which you want the correspondent byte and object providers.
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/api/streaming/StreamingManager.java]: this jdoc is too vague. Here you should explain that this is for the case in which items in an iterable stream contain nested byte streams.

**Comment by ndinu** [comment]
--validate

**Pull request merged**